### DISCUSSION TITLE
Question on data input

### DISCUSSION THREAD

**Discussion by jkissi** [marcomusy/vedo#659] [Q&A] [answered]
Hi Marco!!
Just a quick question here: is there a text input object in vedo? ie a box that the user can write in and that data can be captured? If not, do you have any recommendations for how I might do that?

Cheers
j

**Comment by marcomusy** [comment, marked as answer]
Hi! Interesting question... This is a bit hacky but you can experiment with it:

```python
from vedo import *

settings.enableDefaultKeyboardCallbacks = False

def kfunc(evt):
    global msg
    evt.keyPressed = evt.keyPressed.replace("period", ".")
    if evt.keyPressed == "BackSpace":
        msg = msg[:-1]
    else:
        msg += f"{evt.keyPressed}"
    bu.actor.SetInput(msg)
    plt.render()

def bfunc():
    print("do something")


plt = Plotter(axes=1)
plt.interactor.RemoveObservers("CharEvent") # might be needed

msg = ""
plt.addCallback("key press", kfunc)

bu = plt.addButton(
    bfunc,
    pos=(0.7, 0.05),  # x,y fraction from bottom left corner
    states=["input value here"],
    c=["w"],
    bc=["dg"],  # colors of states
    font="courier",   # arial, courier, times
    size=40,
    bold=True,
    italic=False,
)

mesh = Mesh(dataurl+"magnolia.vtk").c("v").flat()

plt.show(mesh).close()
```

![Screenshot from 2022-06-15 20-58-42](https://user-images.githubusercontent.com/32848391/173904492-81929a42-b413-4c31-ac52-f46ca408a84e.png)

**Reply by jkissi** [reply]
Interesting concept. get it working and am fiddling with it now. Will get back to you with questions (as always!!)

Cheers for the idea!
j
